docs/engine/Life-of-a-Flutter-Frame.md

```diff
@@ -41,6 +40,27 @@ Once a surface is acquired, the [LayerTree][layerTree] is rasterized to the surf
 
 The above process is repeated until the pipeline is empty.
 
+## Warm-up frame
+
+Normally, the Flutter framework begins producing a frame when it receives
+a vsync event from the operating system. However, this may not happen for
+several milliseconds after the app starts (or after a hot reload). To make
+use of the time between when the widget tree is first configured and when
+the engine requests an update, the framework schedules a _warm-up frame_
+using [PlatformDispatcher.scheduleWarmUpFrame][scheduleWarmUpFrame].
+
+A warm-up frame may never actually render (as the engine did not request
+it and therefore does not have a valid context in which to paint), but it
```

Comment:
```suggestion
Depending on the engine implementation, the warm-up frame might cause the first frame to appear slightly earlier
than without, or the warm up frame might not render anything. In either case, it
```

Reply:
I think the "engine context" is not important. And I suggest clarifying the side effect of the first frame time.

Reply:
Hm maybe instead of "engine context" I'll tweak this to say something like:

> ... as it invokes `FlutterView.render` outside of the scope of `PlatformDispatcher.onBeginFrame` or `PlatformDispatcher.onDrawFrame`

That phrasing would be more accurate while also giving some useful context.